Automattic/pocket-casts-android
# Bookmarks: Update feature availability

## Description
This uses feature availability to update the bookmark feature layouts.

## Testing Instructions

Prerequisites:

- Release build
- License test account
- Patron feature enabled
- Bookmark feature enabled  (test for each tier)

1. [Smoke test](https://docs.google.com/spreadsheets/d/1jdTUGVNhhJm6YnXAdUU93QtSvRuY-RW6HSb5bz8Jrd4/edit?usp=sharing) that the appearance of the bookmarks feature correlates with its availability, which is determined by the specific feature tier and user tier.

2. Test that the icon for `Add bookmark` menu on the `Headphone controls` screen is displayed based on the feature tier
(https://github.com/Automattic/pocket-casts-android/pull/1419/commits/df7a8097e391f084a51b53a6005604e273603242) 

    <img width=600 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/84dbbb81-12e1-46ba-b1a1-cd64ce1eb340"/>


## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md N/A
 
#### I have tested any UI changes...

- [x] with different themes N/A
- [ ] with the device set to have a large display and font size N/A
- [ ] for accessibility with TalkBack N/A
